### PR TITLE
Upgrade parquet to 1.8.1

### PR DESCRIPTION
**Pull request by isnotinvain**
This involves porting over the the org.apache.parquet package (from the parquet package), as well as dealing with some private APIs that got changed / removed between 1.6 and 1.8

This is going to be a binary incompatible change given the package rename (the scalding-parquet* sub modules haven't changed much but they deal in 'org.apache.parquet' objects now) so this will trigger a new release version.

### CODE REVIEW

File: scalding-parquet/src/main/scala/com/twitter/scalding/parquet/HasColumnProjection.scala [hunk 1]
```diff
@@ -1,27 +1,67 @@
...
 
-  protected[parquet] final def columnGlobs: Set[ColumnProjectionGlob] = withColumns.map(ColumnProjectionGlob)
+  /**
+   * The format for specifying columns is described here:
+   * https://github.com/apache/parquet-mr/blob/master/parquet_cascading.md#21-projection-pushdown-with-thriftscrooge-records
+   *
+   * Note that the format described there says that multiple globs can be combined with a ; character.
+   * Instead, we use a Set() here and will eventually join the set on the ; character for you.
+   */
+  def withColumnProjections: Set[String] = Set()
```
Comment (isnotinvain) [inline scalding-parquet/src/main/scala/com/twitter/scalding/parquet/HasColumnProjection.scala]: If anyone has a better suggestion to differentiate `withColumns` (deprecated) from `withColumnProjections` (new api) let me know. Could be something shorter like `withColumns2` :(

**Comment by isnotinvain** [comment]
Seems like http://typesafe.artifactoryonline.com/ is down so the builds are failing :/

**Comment by isnotinvain** [comment]
merging https://github.com/twitter/scalding/pull/1381 should fix this

**Comment by johnynek** [comment]
are we ready to bump the monorepo up if we merge this?

**Comment by johnynek** [comment]
or will this block any future publishing of develop until monorepo is upgraded?

**Comment by ianoc** [comment]
It will block future publishing, Alex is going to work on our internal upgrades before this goes in so it doesn't cause issues.

**Comment by isnotinvain** [comment]
Yeah I'm going to get an internal RB ready before we merge this.

**Comment by isnotinvain** [comment]
@johnynek @ianoc @tsdeng the internal RB is posted, so this is ready to merge as well

**Comment by isnotinvain** [comment]
@johnynek @ianoc @tsdeng this is ready to go, end to end tests passed and internal RB is ready as well

**Comment by ianoc** [comment]
LGTM, merge when your ready to do the rls

**Pull request merged**